### ISSUE TITLE
Errors with scheduled backups

### ISSUE THREAD

**Issue by mase76** [jensstein/oandbackup#97]
Hi!
There are cases, that scheduled backups are finished with errors.
But there is no general log to see, where that error was. When
doing single backups or a batched backup I see a summary.
I would suggest to add an option to see the log of the last
(scheduled) backup within the app.
I also suggest to add an option, wether the current backed up
app should be stopped. The error I got was with "OpenVPN for
Android". The file /data/data/de.blinkt.openvpn/cache/mgmtsocket
could not be copied. I am told "No such file or directory. The file is
present, but not be copyable. I filed a bug for openVPN, that such
files should be removed, when the app is not running.

**Comment by jensstein** [comment]
yes, it is absolutely the plan to write a viewer for the log file. for the moment you have to look in the file oandbackup.log in the backup directory.
and do i misunderstand or are you requesting a button to cancel a running backup/restore? that is also on the todo list but will probably not be done in the near future.

to handle these kinds of errors that usually aren't important i could maybe add an option for user-defined error messages to be ignored.

**Comment by DJaeger** [comment]
For the cancel button:
How about a sticky notification with an action button?

**Comment by jensstein** [comment]
a logviewer has now been implemented with 6d2aa0eb9b9e5ebdab6064eee9694423778577e9. you can find it in the menu under tools. it's still very basic.

@DJaeger this issue with a cancel button is not so much the ui implementation as how the ongoing operation should be stopped and the changes rolled back. right now everything is copied directory to its final destination so that would have to change to prevent partially overwriting things.
the actual button could be placed in a sticky notification or on the progress dialog in the program ui.

**Comment by DJaeger** [comment]
@jensstein: How about simply stop after the current apps backup is finished? And I would name it "stop", not "cancel".
With that solution you don't have to reverse anything.

**Comment by jensstein** [comment]
a stop button like that would need a pretty verbose description in the ui to prevent confusion i think. and it would still just be a temporary for a proper cancel function.

**Comment by DJaeger** [comment]
How about creating the new app backups in a folder with attached ```_temp``` to the name and if ready attach a ```_old```to the old one, rename the new one to only the appname, without attachment, and than remove the ```_old``` folder.
If it is canceled, simply remove the ```_temp``` folder.
With that the replacement of the already available data with the new data, would take nearly no time and will always keep the old data until it is really ready.

**Comment by jensstein** [comment]
yes, that is my immediate thought. but then apps with large amounts of data quickly become a problem. what if some app has a data size of 1GB or something and the storage area simply doesn't have enough space for both the temporary files and the original backups.
that may just be the way it is, so maybe there could be a check for available storage and then the cancel functionality could just be disabled on apps with too much data. but if you have another suggestion i'd very much like to hear it.

**Comment by DJaeger** [comment]
The app should anyway check the available storage before starting to backup an app.

There could also be a setting for "safe copying", which enables this way.
So the user can choose between the safe way, which consumes more space or the lesser safe way, which uses lesser space.
The safe way should be the default, as it can also prevent broken backups by crashing or killed app.

If safe way is activated another option for "reverse currend apps backup on cancel" could be available to allow the user to choose the way to cancel (reverse current app or stop after current app).
If safe way is disabled, never use reversing, but stop after current app.

To disable the cancel button is a bad way, as the user have to wait to stop until the next not to big app's backup is started.
As you sometimes simply wants to stop to prevent the heavy battery consuming, it should be available anytime in anyway.
Think of you have a planned backup every night at 3 AM, but one night you are on a party, it is 3:15 AM, you look on your phone and thinks: "Oh, with this battery usage I will not be able to call a taxi if I want to go home, so I should better stop the backup for tonight".

BTW: I think we should move this "stop button" discussion to a new issue and label it as enhancement.

**Comment by jensstein** [comment]
yes, this may be the way to go. i'm still not sold on the idea of a stop button though. but i get your point about the need to stop the ongoing operation somehow.
feel free to open a new issue if you wish to track this :)
